hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionSplitPointRestriction.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.regionserver;
+
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.TableDescriptor;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A split point restriction that restricts the pattern of the split point.
```

Comment:
I think say more about Restriction ... and how they work here and why they can't be done as split policy.

Reply:
... on how a *RegionSplitRestriction differs from a *RegionSplitPolicy?

Reply:
Do we need the 'Point' in the class names?  Isn't RegionSplit enough? Do we need to say RegionSplitPoint? Saying RegionSplit rather than RegionSplitPoint ties these classes to RegionSplitPolicy... which seems good but maybe you don't want that?

Reply:
The split point restriction is what I proposed in HBASE-25706. As here we have two dimentions, one is when do we need to split, such as IncreasingToUpperBound, or ConstantSize, another is how to split, such as KeyPrefix, DelimitedKeyPrefix.

Reply:
Thank you for explaining it! @Apache9
And thank you for reviewing it! @saintstack 

Yes, it seems like we have two dimensions in the current RegionSplitPolicy. And this PR will separate one of the dimensions from RegionSplitPolicy into RegionSplitPointRestriction.

And thank you for the excellent suggestion to rename RegionSplitPointRestriction to RegionSplitRestriction. I will rename it that way. Thanks.